go/pkg/apis/recognizer/devfile_recognizer.go

```diff
@@ -62,6 +66,72 @@ func SelectDevFileUsingLanguagesFromTypes(languages []model.Language, devFileTyp
 	return -1, errors.New("no valid devfile found by using those languages")
 }
 
+func SelectDevFileFromRegistry(path string, url string) (DevFileType, error) {
+	devFileTypes, err := downloadDevFileTypesFromRegistry(url)
+	if err != nil {
+		return DevFileType{}, err
+	}
+
+	index, err := SelectDevFileFromTypes(path, devFileTypes)
+	if err != nil {
+		return DevFileType{}, err
+	}
+	return devFileTypes[index], nil
+}
+
+func downloadDevFileTypesFromRegistry(url string) ([]DevFileType, error) {
+	url = adaptUrl(url)
+	// Get the data
+	resp, err := http.Get(url)
+	if err != nil {
+		// retry by appending index to url
+		url = appendIndexPath(url)
+		resp, err = http.Get(url)
+		if err != nil {
+			return []DevFileType{}, err
+		}
+	}
+	defer resp.Body.Close()
```

Comment:
There are HTTP responses without body are we safe here ?

Reply:
It should be safe 
```
The http Client and Transport guarantee that Body is always non-nil, even on responses without a body or responses with a zero-length body. It is the caller's responsibility to close Body.
```